packages/dcos-integration-test/extra/test_metrics.py

```diff
@@ -0,0 +1,130 @@
+import retrying
+
+LATENCY = 60
+
+
+@retrying.retry(wait_fixed=2000, stop_max_delay=LATENCY * 1000)
+def test_metrics_ping(cluster):
+    """ Test that the metrics service is up.
+    """
+    for agent in cluster.slaves:
+        response = cluster.metrics.get('ping', node=agent)
+        assert response.status_code == 200, 'Got non-200 status code '
+        'from private agent.'
+        assert response.json()['ok'], 'Metrics /ping is not ok on private '
+        'agent.'
+
+    for agent in cluster.public_slaves:
+        response = cluster.metrics.get('ping', node=agent)
+        assert response.status_code == 200, 'Got non-200 status code '
+        'from public agent.'
+        assert response.json()['ok'], 'Metrics /ping is not ok on public agent.'
+
+    for master in cluster.masters:
+        response = cluster.metrics.get('ping', node=master)
+        assert response.status_code == 200, 'Got non-200 status code '
+        'from public agent.'
+        assert response.json()['ok'], 'Metrics /ping is not ok on master.'
+
+
+@retrying.retry(wait_fixed=2000, stop_max_delay=LATENCY * 1000)
+def test_metrics_node(cluster):
+    """Test /system/metrics/api/v0/node endpoint returns
+    correct metrics.
+    """
+    expected_node_metric_datapoints = [
+        'uptime',
+        'processes',
+        'cpu.cores',
+        'load.1min',
+        'load.5min',
+        'load.15min',
+        'cpu.total',
+        'cpu.user',
+        'cpu.system',
+        'cpu.idle',
+        'cpu.wait',
+        'memory.total',
+        'memory.free',
+        'memory.buffers',
+        'memory.cached',
+        'swap.total',
+        'swam.free',
+        'swap.used']
+
+    def expected_datapoint_response(response):
+        for datapoint in response["datapoints"]:
+            if datapoint["name"] not in expected_node_metric_datapoints:
+                return False
```

Comment:
I think the logic is backwards here. This test will return `False` if it finds any datapoints _outside_ the `expected_node_metric_datapoints`, instead of ensuring that the minimum required datapoints exist. Consider the following:

We add a new feature and `response["datapoints"]` contains a datapoint with the `name` "foo". "foo" is not in `expected_node_metric_datapoints`, so this test returns `False` (as expected). But we also want to ensure that a datapoint with the `name` `load.15min` exists. There isn't any logic here to ensure `response["datapoints"]` contains all of the elements of `expected_node_metric_datapoints`.

A better implementation might be:

```python
for datapoint in expected_node_metric_datapoints:
    if datapoint not in response["datapoints"]:
        return False
return True
```

This ensures we're returning the minimum required datapoints, while not breaking this test if we add additional datapoints in the future. Of course, if you want to ensure the minimum required datapoints are there _and_ we haven't added anything new, you could simply check the length of `response["datapoints"]` against the length of `expected_node_metric_datapoints`.